WebRoot/js/zimbraMail/mail/view/prefs/ZmAccountsPage.js

```diff
@@ -318,6 +318,381 @@ function(params) {
 	this._oneTimeCodesDialog.popup();
 };
 
+/*
+ * Recovery Account Section
+ */
+
+/*
+ * Validate Button's Listener.
+ *
+ * Validate code input through the SetRecoveryAccountRequest API Method.
+ *
+ */
+ZmAccountsPage.prototype._handleValidateRecoveryEmailButton =
+	function() {
+	var soapDoc = AjxSoapDoc.create("SetRecoveryAccountRequest", "urn:zimbraMail");
+	var callback = this._recoveryValidateRequest.bind(this);
+	var errorCallback = this._recoveryErrorCallback.bind(this);
+	this.recoveryErrorMessage.innerHTML = "";
+	soapDoc.setMethodAttribute("op", "validateCode");
+	soapDoc.setMethodAttribute("recoveryAccountVerificationCode", this.recoveryCodeInputField.getValue());
+	appCtxt.getAppController().sendRequest({soapDoc: soapDoc, asyncMode: true, callback: callback, errorCallback: errorCallback});
+};
+
+/*
+ * Validate listener callback method.
+ *
+ * Expecting a blank response so the callback acts as confirmation that the code is verified.
+ * Clear the code input field, reset the button, update controls based on "verified" status.
+ *
+ * @param {object} response The response object.
+ */
+ZmAccountsPage.prototype._recoveryValidateRequest =
+function(response) {
+	this.recoveryCodeInputField.setValue("", true);
+	if(!response._isException) {
+		this.recoveryCodeInputField.setValue("", true);
+		this.validateRecoveryEmailButton.setEnabled(false);
+		this._setAccountPasswordControls(this.RECOVERY_EMAIL_STATUS.verified);
+	} else {
+		this._recoveryErrorCallback({code: "Unknown", msg: "Unknown Error: " + ZmMsg.unknownError});
+	}
+};
+
+/*
+ * Reset Button's Listener.
+ *
+ * Reset code and email input through the SetRecoveryAccountRequest API Method.
+ *
+ */
+ZmAccountsPage.prototype._handleResetRecoveryEmailButton =
+	function() {
```

Comment:
Why is this on a new line, but the `appCtxt....` part is beyond what would be a similar line length limit?